torch/lib/c10d/ProcessGroup.cpp

```diff
@@ -62,4 +62,15 @@ ProcessGroup::ProcessGroup(int rank, int size) : rank_(rank), size_(size) {
 
 ProcessGroup::~ProcessGroup() {}
 
+// Q: It might be better not to implement this at all so end users will break
+// build instead of realizing they forgot to implement at a later time?
+// virtual
```

Comment:
Sure, that sounds good. You'll need throwing implementations for the existing backends though.

Reply:
done